File: packages/python/invariance/resources/usage.py
from __future__ import annotations

from ..http_client import HttpClient
from ..types import UsageEvent, UsageQuery


class UsageResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    async def query(self, opts: UsageQuery | None = None) -> list[UsageEvent]:
        return await self._http.get("/v1/usage", params=opts)

File: packages/typescript/src/resources/usage.ts
import type { HttpClient } from '../http.js';
import type { UsageEvent, UsageQuery } from '../types/misc.js';

export class UsageResource {
  constructor(private http: HttpClient) {}

  async query(opts?: UsageQuery): Promise<UsageEvent[]> {
    return this.http.get<UsageEvent[]>('/v1/usage', {
      params: opts as Record<string, string | number | undefined>,
    });
  }
}
